File: java/src/main/java/com/ccpayment/services/MerchantDepositService.java
package com.ccpayment.services;

import com.ccpayment.APIError;
import com.ccpayment.Client;
import com.google.gson.JsonObject;

import java.io.IOException;

public class MerchantDepositService {
    private final Client client;

    public MerchantDepositService(Client client) {
        this.client = client;
    }

    public JsonObject createAppOrderDepositAddress(JsonObject data) throws APIError, IOException {
        return client.post("/createAppOrderDepositAddress", data, JsonObject.class);
    }

    public JsonObject getOrCreateAppDepositAddress(JsonObject data) throws APIError, IOException {
        return client.post("/getOrCreateAppDepositAddress", data, JsonObject.class);
    }

    public JsonObject getAppDepositRecord(String recordId) throws APIError, IOException {
        JsonObject data = new JsonObject();
        data.addProperty("recordId", recordId);
        return client.post("/getAppDepositRecord", data, JsonObject.class);
    }

    public JsonObject getAppDepositRecordList(JsonObject data) throws APIError, IOException {
        return client.post("/getAppDepositRecordList", data, JsonObject.class);
    }
}

File: ruby/lib/ccpayment/services/merchant_deposit_service.rb
module CCPayment
  module Services
    class MerchantDepositService
      def initialize(client)
        @client = client
      end

      def create_app_order_deposit_address(data)
        @client.post('/createAppOrderDepositAddress', data)
      end

      def get_or_create_app_deposit_address(data)
        @client.post('/getOrCreateAppDepositAddress', data)
      end

      def get_app_deposit_record(record_id)
        @client.post('/getAppDepositRecord', { recordId: record_id })
      end

      def get_app_deposit_record_list(data = {})
        @client.post('/getAppDepositRecordList', data)
      end
    end
  end
end
